clients/src/main/java/org/apache/kafka/common/requests/LeaderAndIsrRequest.java

```diff
@@ -35,25 +36,6 @@
 
 public class LeaderAndIsrRequest extends AbstractRequest {
 
-    public static class PartitionState {
```

Comment:
As Grant said, these classes were indeed kept separate because they belong to two separate requests and there is no semantic reason why the data in both would have to be exactly the same. However, we are not following the request data isolation consistently, so it could be OK to share this until they have to diverge.

For that to be possible, the class should not live in a public package. Why not add it as a class in the `requests` package then?